### DISCUSSION TITLE
Upgrade project from Q1 to Q2 - ReferenceErrors

### DISCUSSION THREAD

**Discussion by zanzara** [quasarframework/quasar#10042] [General - Components / Directives / etc]
After following all the instruction to upgrade  my Q1 project into Q2 , I got ReferenceErrors while starting `quasar dev` 
(s. snapshot).
![Bildschirmfoto zu 2021-07-11 13-13-22](https://user-images.githubusercontent.com/7381054/125192789-da6d0580-e249-11eb-90ff-f5530ac106d6.png)

Any ideas how to proceed ?

(there are no errors posted in the terminal)
here is an cut-out of my `quasar.conf.js`
```javascript

 build: {
      vueRouterMode: "hash", // available values: 'hash', 'history'

      // https://quasar.dev/quasar-cli/handling-webpack
      extendWebpack(cfg) {
        cfg.target= "node"    // w/o that I got the error:  'fs' module not installed
      },
      chainWebpack (chain) {
        const nodePolyfillWebpackPlugin = require('node-polyfill-webpack-plugin')
        chain.plugin('node-polyfill').use(nodePolyfillWebpackPlugin)
      }
    },

    // Full list of options: https://quasar.dev/quasar-cli/quasar-conf-js#Property%3A-devServer
    devServer: {
      https: false,
      port: 8080,
      
      //host:'192.168.0.221',
      //allowedHosts: ["lxd-webserver", "localhost:3000", "localhost:8080","192.168.0.168"],  // complains that allowedHosts doesn't exist
      open: "chromium-browser", // true opens browser window automatically
      onBeforeSetupMiddleware(app) {
        const cors = require("cors");
        //app.options("*", cors());   // complains that app.options doesn't exist anymore
        app.use(
          cors({
            allowedHosts: ["lxd-webserver", "localhost:3000", "localhost:8080","192.168.0.168"],
            allowedHeaders: [
              "sessionId",
              "Content-Type",
              "origin",
              "x-requested-with",
              "Authorization",
              "Accept"
            ],
            exposedHeaders: ["sessionId"],
            origin: "*",
            methods: "GET,HEAD,PUT,PATCH,POST,DELETE",
            preflightContinue: false
          })
        );
      }
    },

    // https://quasar.dev/quasar-cli/quasar-conf-js#Property%3A-framework
    framework: {
      iconSet: "material-icons", // Quasar icon set
      lang: "en-US", // Quasar language pack
      config: {},

      importStrategy: "auto",

      // Quasar plugins
      plugins: [
        "LocalStorage",
        "SessionStorage",
        "Notify",
        "Cookies",
        "Dialog",
        "AddressbarColor"
      ]
    },

    // https://quasar.dev/options/animations
    animations: [],

    // https://quasar.dev/quasar-cli/developing-ssr/configuring-ssr
    ssr: {
      pwa: false
    },

```
appreciating any hints or help

**Comment by hawkeye64** [comment]
I'm not sure what's going on, but I'd check your Node version and make sure you were v14.x LTS.

**Reply by zanzara** [reply]
It's 14.17.2

**Reply by hawkeye64** [reply]
and have you tried removing node_modules and the lock file and re-installed them?

**Reply by zanzara** [reply]
nope.
with `npm install`  or `yarn`  ?

**Reply by zanzara** [reply]
Remains the same issue :-( 
after re-installing

**Reply by zanzara** [reply]
any further ideas?  I 'm pinned down and can't go on here

**Reply by hawkeye64** [reply]
This is difficult. Provide your output to `quasar info` and package.json. Make sure @quasar/cli is installed globally and not in your local project. Your local project should contain `quasar`, `@quasar/app`, and `@quasar/extras`.

You can also try creating a new bare-bones project and see if that builds properly. If so, then assume it's something you've added. Slowly add to it from the failing project until you see the same results.

**Comment by zanzara** [comment]
Meanwhile I'd made some more investigations.
The assumption below in `quasar.conf.js` was wrong. I believe it would fix this `fs` module not found error, as I read in a stackoverflow post.
```javascript
extendWebpack(cfg) {
 cfg.target= "node"    // w/o that I got the error:  'fs' module not installed
}
```
It should be (of course guessing, not knowing):  `cfg.target="web" `  as it is default 

But then `quasar dev` is not starting.
After some research I found out, that this [package](https://github.com/nodeca/probe-image-size#readme) `probe-image-size`  needs `fs`  which webpack5 unfortunately doesn't support anymore and the polyfill plugin `node-polyfill-webpack-plugin`  as well not (yet?).
(maybe I'm wrong, but this is how I understood it)

So I remarked out this function for the moment to go on with migration.

Now I got plenty of other errors araising.  :-( 

Question:

- but how could I fix this ` 'fs' module not installed`  error?   
- Should it be fixed by the maintainer of this  `probe-image-size`  package?
- Or do I have to wait until `fs` module is avaliable in the webpack5 polyfill?

(I assume posting my `quasar info` isn't now  useful anymore?! It contains locally this 3 quasar packages you mentioned)

**Reply by hawkeye64** [reply]
That plugin should be using `fs` when building your output, on the Node side of things, not in the browser. Don't get building and running confused. One happens with Node (where fs is available) and one happens in the browser (where fs is not available because browsers don't interact that way for security reasons).

**Reply by zanzara** [reply]
I don't quite understand. If I start `quasar dev` I thought, the webpack is supplying the `devServer `  with  node?!
This plugin isn't called on startup.  It seems to be, that only  the `require` statement is sufficent to get this `fs` error.

Btw.: running my project on quasar v1 everything is fine. In Dev mode and and even productive. So this error must have something to do with the new webpack (imo)

**Reply by hawkeye64** [reply]
This is what I would do: create a bare-bones project. Start adding to it until you see the issue arise.
